### PR TITLE
Remove unnecessary imports of `cupy.linalg`

### PR DESCRIPTION
**Pull request by takagi**
Rel #3420.

This PR removes unnecessary imports of `cupy.linalg` as well as making `cupy/linalg/__init__.py` follow NumPy's reference.

### CODE REVIEW

**Comment by asi1024** [comment]
Could you check the flake8 failure?

File: cupy/__init__.py [hunk 1]
```diff
@@ -59,7 +59,6 @@ def is_available():
 from cupy import functional  # NOQA
 from cupy import indexing  # NOQA
 from cupy import io  # NOQA
-from cupy import linalg  # NOQA
```
Comment (toslunar) [inline cupy/__init__.py:62]: As in `numpy`, the `linalg` submodule should be available via `import cupy; cupy.linalg` or `from cupy import *; linalg`.

Comment (takagi) [reply cupy/__init__.py]: I've fixed it. Thanks!

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**